### PR TITLE
Add tests for diff-highlight

### PR DESCRIPTION
**Pull request by txsmith**

### CODE REVIEW

File: src/projects/diff/diff.html [hunk 1]
```diff
@@ -257,8 +257,8 @@
                 fileUrl: fileUrl,
                 diffId: group[1],
                 changes: group[6],
-                linesChanged: {startLineBefore: group[2], rangeBefore: group[3],
-                               startLineAfter: group[4],  rangeAfter: group[5]}
+                linesChanged: {startLineBefore: group[2], rangeBefore: group[3] | 1,
+                               startLineAfter: group[4],  rangeAfter: group[5] | 1}
```
Comment (txsmith) [inline src/projects/diff/diff.html:262]: Default `range` numbers to `1` if the regex did not match on capturing group 3 or 5.

File: src/projects/diff/highlight.html [hunk 2]
```diff
@@ -90,18 +91,15 @@
           observer: '_render'
         },
         fileUrl: String,
-        collapsed: {
-          type: Boolean,
-          reflectToAttribute: true
```
Comment (txsmith) [inline src/projects/diff/highlight.html:95]: This `reflectToAttribute` was not used

File: test/projects/diff/highlight.html [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+          diffElem.code = diff.changes;
+          diffElem.linesChanged = diff.linesChanged;
+
+          var afterLines = filterEmpty(getChildrenForId('afterColumn'), '');
+          assert.equal(diff.linesChanged.rangeAfter, afterLines.length);
+        });
+
+        test('shows correct amount of empty `before` lines for diff ' + diff.file, function() {
+          diffElem.code = diff.changes;
+          diffElem.linesChanged = diff.linesChanged;
```
Comment (txsmith) [inline test/projects/diff/highlight.html]: I would like to extract this without having to call a function for it in each test. Don't know if this is possble...

Comment (TimvdLippe) [inline test/projects/diff/highlight.html]: Hm foreach, not sure how that works in tests. Could you maybe extract it to a `suite` with a `setUp` where you set the code and linesChanged?

File: src/projects/diff/diff.html (hunk 1 above)
Comment (TimvdLippe) [inline src/projects/diff/diff.html:262]: Maybe add this as a comment in the code with corresponding link to the documentation?

File: test/projects/diff/highlight.html [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+        }
+
+        var filterNonEmpty = function(elems, id) {
+          return elems.filter(function(e) {
+            return e.innerHTML == '';
+          });
+        }
+      });
+    });
+
```
Comment (TimvdLippe) [inline test/projects/diff/highlight.html]: :fire: :fire: :fire: :fire: :fire: :fire: :fire:

**Comment by TimvdLippe** [comment]
Looks good, but it must be rebased :sob:

**Comment by TimvdLippe** [comment]
Aw yeah, party!! :tada:

**Pull request merged**